test/e2e/poseidon_integration.go

```diff
@@ -406,5 +406,114 @@ var _ = Describe("Poseidon", func() {
 			err = framework.WaitForPodRunningInNamespace(clientset, additionalPod)
 			Expect(err).To(HaveOccurred())
 		})
+
+		// Test Nodes does not have any label, hence it should be impossible to schedule Pod with
+		// nonempty Selector set.
+		/*
+			    Testname: scheduler-node-selector-not-matching
+			    Description: Ensure that scheduler respects the NodeSelector field of
+				PodSpec during scheduling (when it does not match any node).
+		*/
+		It("validates that NodeSelector is respected if not matching ", func() {
+			By("Trying to schedule Pod with nonempty NodeSelector.")
+			podName := "restricted-pod"
+			conf := testPodConfig{
+				Name:   podName,
+				Labels: map[string]string{"name": "restricted"},
+				NodeSelector: map[string]string{
+					"label": "nonempty",
+				},
+			}
+			testPod := createTestPod(f, conf)
+			// Clean up additional pod after this test.
+			defer func() {
+				framework.Logf("Time to clean up the pod [%s] now...", testPod.Name)
+				err := clientset.CoreV1().Pods(ns).Delete(testPod.Name, &metav1.DeleteOptions{})
+				Expect(err).NotTo(HaveOccurred())
+			}()
+			err := framework.WaitForPodRunningInNamespace(clientset, testPod)
+			Expect(err).To(HaveOccurred())
+		})
+
+		/*
+			    Testname: scheduler-node-selector-matching
+			    Description: Ensure that scheduler respects the NodeSelector field
+				of PodSpec during scheduling (when it matches).
+		*/
+		It("validates that NodeSelector is respected if matching ", func() {
+			// Randomly pick a node
+			nodeName := getNodeThatCanRunPodWithoutToleration(f)
+
+			By("Trying to apply a random label on the found node.")
+			k := fmt.Sprintf("kubernetes.io/e2e-%d", rand.Uint32())
+			v := "42"
+			nodeList, err := clientset.CoreV1().Nodes().List(metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			for _, node := range nodeList.Items {
```

Comment:
@m1093782566 Are you trying to apply the label on a single node or all the nodes?
It will loop and apply labels on all the nodes.
And in [Line 489](https://github.com/m1093782566/poseidon/blob/594ee450abf6b2a860c8f6b835842eaac60f1162/test/e2e/poseidon_integration.go#L489) you are expecting it to match the same nodename which you ramdonly selected.
This is bug, and the failure is random.

I ran three times on my local gce cluster, twice it passed and once it failed.
So please remove the ```for loop``` apply the label to the node found in [line 445](https://github.com/m1093782566/poseidon/blob/594ee450abf6b2a860c8f6b835842eaac60f1162/test/e2e/poseidon_integration.go#L445)